### DISCUSSION TITLE
Is there a way to use something like Profile.Repository in a template?

### DISCUSSION THREAD

**Discussion by 2q2code** [creativeprojects/resticprofile#436] [Q&A]
I would like to reference the basename or path of the repository being operated on so that I can request the encryption password for that repository from the secrets system "Infisical". So when I run a command to get the password for the repository, I can pass the repository path or part thereof as an argument or environment variable to the script. I scanned the docs but was unable to see the repository being set in an environment variable except possibly when passed to the restic binary.

**Comment by creativeprojects** [comment]
Would you be able to add a password entry based on the profile name?

I don't mind adding the repository variable though, although it would have to be redacted (like user & password removed from a SFTP url)

**Reply by 2q2code** [reply]
I have been using the profile name, and I have actually found a workaround, so it's okay for now, but it just seemed like then file path or basename of the repository would make sense to use as a lookup especially if you have multiple profiles backing up to the one repo.
